http/handler/handler.go

```diff
@@ -41,7 +41,7 @@ type Handler struct {
 	hostHandler            *host.Handler
 	pingHandler            *ping.Handler
 	securedProtocol        bool
-	edgeManager            *edge.Manager
+	edgeManager            *edge.Manager // TODO: I suspect we should not store this here
```

Comment:
where else?

Reply:
we store pointers to core objects all over the place - that's a way to hide the fact your code has lots of singletons, and creates permanent interdependencies where often they're not needed.

Reply:
See the last commit for an attempt at not storing a pointer to the EdgeManager in the Handler.